File: my-folder/0070-climbing-stairs/solution.java
import java.util.HashMap;

class Solution {
    HashMap<Integer, Integer> memo = new HashMap<>();
    
    public int climbStairs(int n) {
        if (memo.containsKey(n)) {
            return memo.get(n);
        }  else if (n == 1) {
            return 1;
        } else if (n == 0) {
            return 1;
        } else {
            int ans = climbStairs(n - 1) + climbStairs(n - 2);
            memo.put(n, ans);
            return ans;
        }
    }
}

File: my-folder/0070-climbing-stairs/solution.py
class Solution(object):

    def __init__(self):
        self.answers = {}

    def climbStairs(self, n):
        if n == 0:
            return 1
        if n == 1:
            return 1
        if n in self.answers.keys():
            return self.answers[n]
        answer = self.climbStairs(n - 1) + self.climbStairs(n-2)
        self.answers[n] = answer
        return answer

        """
        :type n: int
        :rtype: int
        """
        if n == 0:
            return 1
        if n == 1:
            return 1
        return self.climbStairs(n - 1) + self.climbStairs(n-2)
